src/components/PrivateRoute.js

```diff
@@ -0,0 +1,29 @@
+import React, { useContext } from "react";
+import { Route, Redirect } from "react-router-dom";
+import { AuthContext } from "../services/providers/authProvider";
+
+/**
+ * This component masks the specified route to users that are not authenticated through
+ * the firebase application and redirects them to the login screen
+ * @param {*} param0 The component that will be shown to the user if they are logged in
+ * @returns ...
+ */
+const PrivateRoute = ({ component: RouteComponent, ...rest }) => {
+  const {currentUser} = useContext(AuthContext);
+  return (
+    <Route
+      {...rest}
+      render={routeProps =>
+        !!currentUser ? (
```

Comment:
Is this a double negation?

Reply:
!! casts variables to boolean. If currentUser is null it would be false